zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/listeners/task/TaskListenerTest.java

```diff
@@ -0,0 +1,496 @@
+    final long processInstanceKey =
+        createProcessInstance(
+            createProcess(
+                p ->
+                    p.userTask(
+                            USER_TASK_ELEMENT_ID,
+                            t ->
+                                t.zeebeUserTask()
+                                    .zeebeAssignee("foo")
+                                    .zeebeTaskListener(l -> l.complete().type(LISTENER_TYPE)))
+                        .serviceTask(
+                            "subsequent_service_task",
+                            tb -> tb.zeebeJobType("subsequent_service_task"))));
+
+    ENGINE.userTask().ofInstance(processInstanceKey).complete();
+
+    // when: complete TL job with a variable 'my_listener_var'
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(LISTENER_TYPE)
+        .withVariable("my_listener_var", "bar")
+        .complete();
+
+    // then: assert the variable 'my_listener_var' is not accessible in the subsequent element
+    final var subsequentServiceTaskJob = activateJob(processInstanceKey, "subsequent_service_task");
+    assertThat(subsequentServiceTaskJob.getVariables()).doesNotContainKey("my_listener_var");
+    completeJobs(processInstanceKey, "subsequent_service_task");
+  }
+
+  @Test
+  public void shouldAllowTaskListenerVariablesInUserTaskOutputMappings() {
+    // given: deploy process with a user task having complete TL and service task following it
+    final long processInstanceKey =
+        createProcessInstance(
+            createProcess(
+                p ->
+                    p.userTask(
+                            USER_TASK_ELEMENT_ID,
+                            t ->
+                                t.zeebeUserTask()
+                                    .zeebeAssignee("foo")
+                                    .zeebeTaskListener(l -> l.complete().type(LISTENER_TYPE))
+                                    .zeebeOutput("=my_listener_var+\"_abc\"", "userTaskOutput"))
+                        .serviceTask(
+                            "subsequent_service_task",
+                            tb -> tb.zeebeJobType("subsequent_service_task"))));
+
+    ENGINE.userTask().ofInstance(processInstanceKey).complete();
+
+    // when: complete TL job with a variable 'my_listener_var'
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(LISTENER_TYPE)
+        .withVariable("my_listener_var", "bar")
+        .complete();
+
+    // then: assert the variable 'userTaskOutput' is accessible in the subsequent element
+    final var subsequentServiceTaskJob = activateJob(processInstanceKey, "subsequent_service_task");
+    assertThat(subsequentServiceTaskJob.getVariables()).containsEntry("userTaskOutput", "bar_abc");
+    completeJobs(processInstanceKey, "subsequent_service_task");
+  }
+
+  @Test
+  public void shouldIncludeConfiguredUserTaskDataInCompleteTaskListenerJobHeaders() {
```

Comment:
💭 This is fine for now. It's possible that we remove this later again if we support explicit user data on the job record. While that's still up to debate, I think it's great that we already support some form of providing this data to the listeners.